domain/model/state/state.go

```diff
@@ -369,13 +345,30 @@ AND removed = false
 		return fmt.Errorf("getting user uuid for setting model %q owner: %w", input.Name, err)
 	}
 
+	// deleteBadStateModel is here to allow models to be recreated that may have
+	// failed during the full model creation process and never finalised. We
+	// will only ever allow this to happen if the model is not finalised.
+	deleteBadStateModel := `
```

Comment:
Do we also need to delete things like model agent or model credential as well? Can they be left behind from a previous failed creation?

Reply:
Yes we do, I will work on this logic with Simon in a follow up PR. This is just the start of getting that work going. It will be another big one to get it all ironed out.